### ISSUE TITLE
Question: Terminal IDE for scripting in Lanterna

### ISSUE THREAD

**Issue by ddovnar** [mabe02/lanterna#527]
Hi! It is possible to create a terminal-base IDE using lanterna? 
I need script editor with syntax highlight 
Some properti editor panel and several settings
Thank's

**Comment by mabe02** [comment]
Yes, I don't see why not? You'll probably need to re-implement the TextBox component as the main source editor because the bundled one was never designed to support syntax highlighting.

**Comment by panicz** [comment]
FWIW I've been implementing an IDE for coding in Lisp-like (i.e. fully parenthesized) languages.
I currently have 3 clients, one of them uses Lanterna for input and rendering.

Neither client is usable yet (I'm hoping to make it usable by the end of the year), but you can track the progress here, and even fiddle around with it a bit:
https://github.com/panicz/grasp
